webExtensionApis/study/src/studyUtils.js

```diff
@@ -0,0 +1,758 @@
+   * startup for the reason ADDON_INSTALL. For more on study states like 'enter'
+   * see ABOUT.md at github.com/mozilla/shield-studies-addon-template
+   *
+   * Side effects:
+   * - sends 'enter'
+   * - sets this._internals.prefs.firstRunTimestamp to Date.now()
+   *
+   * @returns {void}
+   */
+  async firstSeen() {
+    this.throwIfNotSetup("firstSeen uses telemetry.");
+    logger.debug(`attempting firstSeen`);
+    this._internals.isFirstRun = true;
+    await this._telemetry({ study_state: "enter" }, "shield-study");
+    this.setFirstRunTimestamp("" + Date.now());
+  }
+
+  /**
+   * Marks the study's telemetry pings as being part of this experimental
+   * cohort in a way that downstream data pipeline tools
+   * (like ExperimentsViewer) can use it.
+   * @returns {void}
+   */
+  setActive() {
+    this.throwIfNotSetup("setActive uses telemetry.");
+    const info = this.info();
+    logger.debug(
+      "marking TelemetryEnvironment",
+      info.activeExperimentName,
+      info.variation.name,
+    );
+    TelemetryEnvironment.setExperimentActive(
+      info.activeExperimentName,
+      info.variation.name,
+    );
+  }
+
+  /**
+   * Removes the study from the active list of telemetry experiments
+   * @returns {void}
+   */
+  unsetActive() {
+    this.throwIfNotSetup("unsetActive uses telemetry.");
+    const info = this.info();
+    logger.debug(
+      "unmarking TelemetryEnvironment",
+      info.activeExperimentName,
+      info.variation.name,
+    );
+    TelemetryEnvironment.setExperimentInactive(info.activeExperimentName);
+  }
+
+  /**
+   * Adds the study to the active list of telemetry experiments and sends the
+   * "installed" telemetry ping if applicable
+   * @param {string} reason - The reason the addon has started up
+   * @returns {void}
+   */
+  async startup() {
+    this.throwIfNotSetup("startup");
+    const isFirstRun = this._internals.isFirstRun;
+    logger.debug(`startup.  setting active. isFirstRun? ${isFirstRun}`);
+    this.setActive();
+    if (isFirstRun) {
+      await this._telemetry({ study_state: "installed" }, "shield-study");
+    }
+  }
+
+  /**
+   * @async
+   * Ends the study:
+   *  - Removes the study from the active list of telemetry experiments
+   *  - Sends a telemetry ping about the nature of the ending
+   *    (positive, neutral, negative)
+   *  - Sends an exit telemetry ping
+   * @param {string} endingName - The reason the study is ending, see
+   *    schema.studySetup.json
+   * @returns {Object} endingReturned _internals.endingReturned
+   */
+  async endStudy(endingName) {
+    this.throwIfNotSetup("endStudy");
+
+    // also handle default endings.
+    const alwaysHandle = ["ineligible", "expired", "user-disable"];
+    let ending = this._internals.studySetup.endings[endingName];
+    if (!ending) {
+      // a 'no-action' ending is okay for the 'always handle'
+      if (alwaysHandle.includes(endingName)) ending = {};
```

Comment:
Hm is there a reason why this isn't using braces? I'm a fan of always using them for `if` statements in general (since it's so easy to mistake indent for scope and miss the fact that there's no brace, when adding a new case to existing code) but I haven't seen this style of `if`/`else` before I don't think.

Reply:
(no reason, fixing.  Seems like that should be lintable. 

https://eslint.org/docs/rules/curly  

Not in mozilla-recommended or eslint-recommended, happy to add it.